src/Dotnet.Script.DependencyModel.Nuget/NuGetSourceReferenceResolver.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+
+namespace Dotnet.Script.DependencyModel.NuGet
+{
+    /// <summary>
+    /// A <see cref="SourceReferenceResolver"/> decorator that handles
+    /// references to NuGet packages in scripts.  
+    /// </summary>
+    public class NuGetSourceReferenceResolver : SourceReferenceResolver
+    {
+        private readonly SourceReferenceResolver _sourceReferenceResolver;
+        private readonly IDictionary<string, IReadOnlyList<string>> _scriptMap;
```

Comment:
since this is part of a public API I think for easier comprehension we should wrap it into a class rather than a fairly cryptic `IDictionary<string, IReadOnlyList<string>>`

Reply:
I agree. The reason I did that here was to keep the number of dependencies on a low down for the `Dotnet.Script.DependencyModel.NuGet` package. Since we bring back a reference to this assembly as a hack inside the `NuGetMetadataReferenceResolver`. If we reference `Dotnet.Script.DependencyModel` from this package, I am afraid that we bring in types from that assembly too. Not sure if that is an issue though. I'll take a look at it

Reply:
maybe let's leave it as is for now. once we have an idea of how to address https://github.com/filipw/dotnet-script/pull/179#discussion_r152201758 this should not be needed anymore, as the map will not be passed into the `NuGetSourceReferenceResolver` anymore, it will build itself up from within